usecases/vessel-detection/index.md

```diff
@@ -0,0 +1,171 @@
+# Vessel Detection
+
+In this notebook we will learn how to apply adaptive thresholding to SENTINEL1_GRD data within OpenEO Platform. We will then take a look at our results against vessel location data from the Maritime Traffic Agency over the Adriatic. We will be using the the openeo-python-client to prepare our process graph, and a Plotly Dash dashboard to interact with our results.
+
+### Adaptive Thresholding in Xarray
+
+After investigating the code initially provided by Planetek, we arrived at this xarray implementation of adaptive thresholding.
+
+```python
+WINDOW_LAT_SIZE = 30
+WINDOW_LON_SIZE = 30
+THRESHOLD_FACT = 4
+out = xr.ones_like(data)
+rolling_mean = data.rolling(
+    longitude=WINDOW_LAT_SIZE, latitude=WINDOW_LON_SIZE, center=True
+).mean()
+thresholded_image = data > rolling_mean * THRESHOLD_FACT
+raster = out.where(thresholded_image == True, other=0)
+```
+
+We are going to port this into an OpenEO process graph.
+
+### 1. Authenticate
+
+
+```python
+import openeo
+
+backend = "openeo.cloud"
+conn = openeo.connect(backend)
+conn = conn.authenticate_oidc()
+```
+
+### 2. Pre-processing prep
+
+This process graph requires either a sea or land mask to remove the land from our AOI. A seamask has been prepared for this example and is available on github. It was derived using the [Water Bodies](https://land.copernicus.eu/content/corine-land-cover-nomenclature-guidelines/html/) classes from Corine Land Cover.
+
+
+```python
+import folium
+
+# We are going to use the following GeoJson to operate as a sea mask for out process graph.
+SEA_MASK = "https://raw.githubusercontent.com/SerRichard/sea_mask/main/sea-mask-4326.json"
+
+fig = folium.Figure(width=600, height=400)
+map = folium.Map(location=[44.465488, 12.602316], zoom_start=8)
+folium.GeoJson(SEA_MASK).add_to(map)
+fig.add_child(map)
+```
+![SeaMask to be used](./seamask.png)
+
+### 3. Defining a process graph to detect vessels in our AOI.
+
+The spatial_extent we have defined includes, but is not limited to the sea mask that we will be using. We will use both available polarizations from the SENTINEL1_GRD collection, and run this graph on a little over a weeks worth of data.
+
+
+```python
+spatial_extent  = {
+          "west": 12.194377989297493,
+          "east": 12.758093271633888,
+          "south": 44.24420099164355,
+          "north": 44.85455845353388
+        }
+
+temporal_extent = [
+          "2021-10-01",
+          "2021-10-09"
+        ]
+
+collection      = "SENTINEL1_GRD"
```

Comment:
note that most algorithms don't work on raw GRD and require a 'sar_backscatter' process. 
I don't know about this one however, so could be fine.

A more cosmetic suggestion could be to avoid defining collection and bands as separate variables, as they are not reused anyway. (Makes the code shorter, which is always good to give users the feeling that they can do a lot with limited code.)

Reply:
Yes, it uses the raw GRD data.

Updated in line with collection id and bands comment!